File: Demo14_ObjC_ARC/Demo14/Driver.m
//
//  Driver.m
//  Demo14
//
//  Created by 曾敏傑 on 2019/11/25.
//  Copyright © 2019 mjz. All rights reserved.
//

#import "Driver.h"

@implementation Driver
-(id)init{
    self = [super init];
    NSLog(@"Driver init");
    if(self){
        _name = @"CHT auto pilot";
    }
    return self;
}
-(void) dealloc{
    NSLog(@"Driver will be relaesed");
}

@end

File: Demo15_ios_memory_leak/Demo15/Driver.swift
//
//  Driver.swift
//  Demo15
//
//  Created by 曾敏傑 on 2019/11/25.
//  Copyright © 2019 mjz. All rights reserved.
//

import Foundation
class Driver{
    let name:String
    init(name:String){
        self.name = name
        print("Driver \(name) is initialized")
        
    }
    deinit{
        print("deinit Driver \(name)")
    }
    var ownedCar:Car?
}
